ledger/src/blockstore.rs

```diff
@@ -1426,6 +1459,8 @@ impl Blockstore {
         data_index: &mut ShredIndex,
         shred: &Shred,
         write_batch: &mut WriteBatch,
+        is_recovered: bool,
+        is_repaired: bool,
```

Comment:
Two `bool` arguments next to each other lacks type-safety and might be bug prone.
Also, I am guessing these are mutually exclusive. If so might be good to define some enum here like:
```
enum ShredSource {
   UNSPECIFIED,
   TURBINE,
   RECOVERED,
   REPAIRED,
   ....
}
```